libs/rest/src/struct/Bucket.ts

```diff
@@ -77,47 +75,29 @@ export class Bucket {
     if (limit) state.limit = Number(limit);
     if (remaining) state.remaining = Number(remaining);
     if (resetAfter) state.timeout = Number(resetAfter) * 1000;
+    this.rest.emit('response', req, res.clone(), state);
 
     await this.mutex.set(this.route, state);
 
     if (res.status === 429) {
-      const retry = res.headers.get('Retry-After');
+      const retry = res.headers.get('retry-after');
       /* istanbul ignore next */
-      const retryAfter = retry ? Number(retry) * 1000 : 0;
+      const retryAfter = Number(retry ?? 1) * 1000;
 
-      this.manager.emit('ratelimit', this.route, req.path, false, retryAfter);
+      this.rest.emit('ratelimit', this.route, req.path, false, retryAfter);
 
       await this.mutex.set(this.route, { timeout: retryAfter });
-      return this._retry<T, D, Q>(req);
+      return Promise.reject(new CordisRestError('rateLimited'));
```

Comment:
This function is async, why are we returning another Promise object with the rejection over throwing??

Reply:
It's just a practice I've taken over time due to [some particular scenarios](https://stackoverflow.com/questions/33445415/javascript-promises-reject-vs-throw).

Technically we don't have that at all in our codebase so I can't really argue for consistency - but "returning a new promise" isn't particularly accurate nor damaging with how Node async works and [a lot of codebases](https://github.com/discordjs/discord.js/blob/3c175cb5116fe50ba3084163565dd244a25b657f/src/structures/GuildChannel.js#L233) tend to do it.